From 6102bb628bf67fff5ebd883731aac921fde20dc9 Mon Sep 17 00:00:00 2001
From: Shurisko <thomas.bellettini@epitech.eu>
Date: Fri, 16 Jun 2023 13:01:29 +0200
Subject: [PATCH] Patch meta


diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 05d14bf26..ddce2580f 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -207,7 +207,10 @@ public class CraftEventFactory {
 
         ItemMeta meta = itemInHand.getItemMeta();
 
-        Bukkit.getLogger().info("Je suis trigger ! [" + meta.getDisplayName() + "]");
+        if (meta.getIdentifier() == null) {
+            meta.setIdentifier(UUID.randomUUID().toString());
+        }
+        Bukkit.getLogger().info("Je suis trigger ! [" + meta.getIdentifier() + "]");
 
         if (position.getY() > 255) {
             blockClicked = null;
-- 
2.40.1.windows.1

